### DISCUSSION TITLE
Unable to proxy another internal private proxy

### DISCUSSION THREAD

**Discussion by vicnguyenqt** [chimurai/http-proxy-middleware#880] [Q&A]
Hi,
I've been trying to create a proxy that redirects traffic to another reserved proxy. 
But the thing is that internal reserved proxy has an **IP restriction**, eventually, when forward the traffic, here is the log:
`2023-02-21 14:22:00:220 - error: Proxy forward error:  getaddrinfo ENOTFOUND pkg-debian-staging 
Feb 21 06:22:00 [debian-proxy-staging] [app/web.1]   {"errno":-3008,"code":"ENOTFOUND","syscall":"getaddrinfo","hostname":"pkg-debian-staging","stack":"Error: getaddrinfo ENOTFOUND pkg-debian-staging \n    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26)"}`
To my understanding, seems that [http-proxy](https://www.npmjs.com/package/http-proxy) tries to request/validate the URL first, and throw error to the event emitter.
How could I send, e.g: my IP address to header/ bypass that error? I couldn't open that internal proxy (prohibited)

**Comment by chimurai** [comment]
Did you try to configure your internal proxy?

https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/corporate-proxy.md

**Reply by vicnguyenqt** [reply]
Yup, I did try that one, but looks like it return another error. Didn’t remember correctly, but something like: Cannot set header before request…
I wrote by TypeScript. By importing the module, got Type error…

**Reply by vicnguyenqt** [reply]
FYi: Everything looks good in localhost, but deployed to stage/prod seems the forwarding traffic took the CDN (domain) IP address.
